### ISSUE TITLE
FR: Image resizing

### ISSUE THREAD

**Issue by quesada** [zim-desktop-wiki/zim-desktop-wiki#1423]
Right now, pasting an image keeps the original size, which could be inappropriate for a notetaking program

**Comment by jiewangjs** [comment]
strongly agree. Especially in HiDPI machine, this is a very unconvient thing. please give a resize drag feature soon.

**Comment by jEsuSdA** [comment]
I think an usable approach will be to fit image with to zim width (and of course, the height should be the one to avoid image deformation).

Maybe a configuration value to fill the percentaje of zim with to scale images will be incredible good solution. So user can force a 100% width or 50% width (to have two images for row), etc.

As a more accuracy solution, launching the image viewer at clicking image will be a good point.

What do you think about?

**Comment by fdt-dev** [comment]
Right-clicking on an image will bring up a dialogue window where the picture can be resized. In the same dialogue window a link can be added. It's all very simple and a lot of the ideas already are implemented. Have you tried that?

It's just that Zim can't do automatic resizing, as far as I understand how Zim works. I guess that must have to do with the fact that the "page" doesn't have a fixed length or width. Zim isn't a word-processor.
I just want pitch in here because I am pretty happy with how Zim currently deals with pictures. I think it's great how it works, resizing is easy and accurate and adding links to pictures (so you can click them to open them in the standard image viewer) is also perfectly straightforward. Just voicing my opinion.

**Comment by neinseg** [comment]
> Right-clicking on an image will bring up a dialogue window where the picture can be resized. In the same dialogue window a link can be added. It's all very simple and a lot of the ideas already are implemented. Have you tried that?

I had the same issue as OP just now, and I had to search through this issue tracker to find out that these options are hidden under "edit properties" in the context menu. Discoverability of this feature is poor, and it breaks with the UI convention used with other text editing programs that allow images to be inserted.

**Comment by jEsuSdA** [comment]
> Right-clicking on an image will bring up a dialogue window where the picture can be resized. In the same dialogue window a link can be added. It's all very simple and a lot of the ideas already are implemented. Have you tried that?
> 

I just discover that. Thanks a lot, @fdt-dev !!! ;)

**Comment by directentis1** [comment]
Can we add diagram's property to resize diagram images  generated inside Zim?
Currently it has only "edit" option just to  edit code snippets and regenerate those diagrams.

![](https://i.imgur.com/tUbSpdn.png)

**Comment by LSeelig** [comment]
Is it possible one could set a width of editor as part of the page template or editor settings, and images would be displayed inside of the width? This would be appreciated for both text and images and might be a rather elegant solution.
Images with less width than the column would still display at their native res.

My workflow, which involves text notes for most classes, but occasional images for slides. In mathematics classes, I take my notes with a stylus. The result are files with a width that matches the ipad's vertical resolution and an extensive height. If the width simply conformed to the width of the text in the editor, the pages would display perfectly. I would still have to convert them from svg to png to display, but that's a separate issue.

**Comment by introt** [comment]
See also:
- Fit images inside viewport #324 
- FR: User-defined default pasted image format #1937 (includes size)

**Comment by castorfou** [comment]
with our high resolution screens these days

when I screnshot+paste images are then much bigger
don't know if we could use the scale parameter from the system 
![image](https://github.com/user-attachments/assets/ecad6553-d62a-4ebd-9885-c6cabd944599)

to get identical size in zim to what is on screen

**Comment by jaap-karssenberg** [comment]
You can edit the display size of the image manually. Unfortunately there is
no way to know the original display size for Zim

Op di 31 dec. 2024 12:42 schreef Guillaume Ramelet ***@***.***

**Comment by webdevelopers-eu** [comment]
TLDR; **The optional setting should limit the width of the image only if it exceeds its natural size. This solution would address all the issues for me and appears to be the simplest to implement compared to other suggestions I have encountered.**

* When using external monitors, automatic adjustment to fit the viewport is not ideal. Images pasted on large external monitors become scrollable when viewed on the notebook screen.
* Pasting images at their natural size is also problematic, as screenshots will inevitably result in overflow in 100% of cases. A screenshot of the screen pasted into a smaller window will always require right-clicking to manually adjust the size.

Automatic resizing logic is inherently flawed and bound to fail in large portion of cases. The best solution would be a straightforward option: "Default image size: automatic/fixed max-width." Currently, I find myself forced to manually right-click and adjust the width of every pasted image, which unconsciously discourages me from pasting images altogether, especially under time pressure. This results in a frustrating user experience, arguably the worst possible UX for this scenario.

**Comment by raacampbell** [comment]
On Confluence wikis the user has the option to show in-line images as a one of a selection of down-scaled sizes. Clicking on the image brings up the original size temporarily. This is very useful. A screenshot might well contain details that are only visible in the full size view, but having the full size image always in-line is not practical.

**Comment by ARoxdale** [comment]
Though the OP question revolves around pasting in images, on the general topic of "resizing images", the feature request I would make is to enable some kind of easy click and drag resizing of images, as can be done in OneNote or Xournal++.

I am aware that a resize dialog is available. However as someone who uses a lot of images in notes, this would be cumbersome to use. A faster, more immediate mouse based resize (perhaps with keyboard shortcuts on image highlight), would probably solve a lot of the various issues people are having with image import. If a resize is one mouse click and drag away, then the decisions about size on import are not as important anymore.

I don't know how difficult click and drag resize is in GtK. Is this something that could be handled by a plugin?
